Gems/EMotionFX/Code/MCore/Source/StringConversions.h

```diff
@@ -50,18 +41,6 @@ namespace MCore
 namespace AZStd
 {
     void to_string(string& str, bool value);
```

Comment:
nit: Why not lift all the conversions out?

Reply:
The bool conversion specifically breaks AssetProcessor for reasons that are beyond me. I'm consequently keeping it scoped.

Reply:
The bool conversion is gross. It adds an overload that has significant impact on overload resolution. Consider these
overloads:

```cpp
void to_string(AZStd::string& dest, AZStd::wstring_view src);
void to_string(string& str, bool value);
```

And then calling code like this:
```
WCHAR src[260];
AZStd::string dst;
AZStd::to_string(dst, src); // Which overload does this call?
```

If the .cpp has not included `MCore/Source/StringConversions.h`, the call
to `to_string()` will convert the `WCHAR[260]` type to a
`AZStd::wstring_view`, and call the first overload. But if
`StringConversions.h` _has_ been included, the implicit conversion of
`WCHAR[260]` to `bool` has a higher precedence, and it will be chosen
instead.

This overload was causing some uses of `to_string` in
`AnimGraph/GameController.cpp` to resolve to the wrong overload in unity
builds. I have a commit that I made this morning to remove it completely.

Reply:
We can move this to AzCore/Math as well.
We just need to use a constraint that the `bool` overload accepts a type that is exactly `bool` once cv ref qualifiers have been removed
https://godbolt.org/z/G3ce383T1
```
template<class BoolType>
auto to_string(std::string& dest, BoolType value) ->
    std::enable_if_t<std::same_as<std::remove_cvref_t<BoolType>, bool>>
```

Reply:
I'm going to pull the to_string changes out into a separate PR. Now that I'm aware of AzCore/Math/ToString, I think it's warranted.